### PR TITLE
Extract 3 extra packages and repositories from FLP prototype

### PR DESCRIPTION
**Pull request by Barthelemy**
- _Common-O2_ - repo for common utilities used in several repos
- _InfoLogger_ 
- _RORC_

### CODE REVIEW

**Comment by Barthelemy** [comment]
@ktf @dberzano Ready for review

File: common-o2.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#!/bin/sh
+
+case $ARCHITECTURE in
+  osx*) BOOST_ROOT=$(brew --prefix boost) ;;
+esac
+
+cmake $SOURCEDIR                                              \
+      -DCMAKE_INSTALL_PREFIX=$INSTALLROOT                     \
+      -DCMAKE_PREFIX_PATH=$PROTOBUF_ROOT                      \
+      ${BOOST_ROOT:+-DBOOST_ROOT=$BOOST_ROOT}                 \
```
Comment (dberzano) [inline common-o2.sh]: Do we really need `-DBOOST_ROOT` plus `-DBoost_DIR` and `-DBoost_INCLUDE_DIR`? Normally the second is sufficient. Also, the check should be on `BOOST_VERSION` and not `BOOST_ROOT`.

File: common-o2.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+cat > etc/modulefiles/$PKGNAME <<EoF
+#%Module1.0
+proc ModulesHelp { } {
+  global version
+  puts stderr "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+}
+set version $PKGVERSION-@@PKGREVISION@$PKGHASH@@
+module-whatis "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+# Dependencies
+module load BASE/1.0 ${BOOST_VERSION:+boost/$BOOST_VERSION-$BOOST_REVISION} ${GCC_TOOLCHAIN_ROOT:+GCC-Toolchain/$GCC_TOOLCHAIN_VERSION-$GCC_TOOLCHAIN_REVISION}
```
Comment (dberzano) [inline common-o2.sh]: `GCC_TOOLCHAIN_ROOT` should be `GCC_TOOLCHAIN_VERSION` to be consistent with other recipes (the `BOOST_*` part is OK).

File: infologger.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+proc ModulesHelp { } {
+  global version
+  puts stderr "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+}
+set version $PKGVERSION-@@PKGREVISION@$PKGHASH@@
+module-whatis "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+# Dependencies
+module load BASE/1.0 
+            ${BOOST_VERSION:+boost/$BOOST_VERSION-$BOOST_REVISION}
+            ${GCC_TOOLCHAIN_ROOT:+GCC-Toolchain/$GCC_TOOLCHAIN_VERSION-$GCC_TOOLCHAIN_REVISION}
```
Comment (dberzano) [inline infologger.sh]: `GCC_TOOLCHAIN_VERSION` instead of `ROOT`.

File: monitoring.sh [hunk 4]
```diff
@@ -22,11 +22,11 @@ esac
 
 cmake $SOURCEDIR                                              \
       -DCMAKE_INSTALL_PREFIX=$INSTALLROOT                     \
-      -DCMAKE_PREFIX_PATH=$PROTOBUF_ROOT                      \
       ${BOOST_ROOT:+-DBOOST_ROOT=$BOOST_ROOT}                 \
       ${BOOST_ROOT:+-DBoost_DIR=$BOOST_ROOT}                  \
       ${BOOST_ROOT:+-DBoost_INCLUDE_DIR=$BOOST_ROOT/include}  \
-      ${APMON_CPP_ROOT:+-DAPMON_ROOT=$APMON_CPP_ROOT}
+      ${APMON_CPP_ROOT:+-DAPMON_ROOT=$APMON_CPP_ROOT}         \
```
Comment (dberzano) [inline monitoring.sh]: This check and all the BOOST ones above should be on `_VERSION` not `_ROOT`.

File: readoutcard.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+---
+#!/bin/sh
+
+case $ARCHITECTURE in
+  osx*) BOOST_ROOT=$(brew --prefix boost) ;;
+esac
+
+cmake $SOURCEDIR                                              \
+      -DCMAKE_INSTALL_PREFIX=$INSTALLROOT                     \
+      ${BOOST_ROOT:+-DBOOST_ROOT=$BOOST_ROOT}                 \
```
Comment (dberzano) [inline readoutcard.sh]: Again :-)

File: readoutcard.sh [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+proc ModulesHelp { } {
+  global version
+  puts stderr "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+}
+set version $PKGVERSION-@@PKGREVISION@$PKGHASH@@
+module-whatis "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+# Dependencies
+module load BASE/1.0 
+            ${BOOST_VERSION:+boost/$BOOST_VERSION-$BOOST_REVISION}
+            ${GCC_TOOLCHAIN_ROOT:+GCC-Toolchain/$GCC_TOOLCHAIN_VERSION-$GCC_TOOLCHAIN_REVISION}
```
Comment (dberzano) [inline readoutcard.sh]: ...and again :-)

File: common-o2.sh [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  - boost
+  - "GCC-Toolchain:(?!osx)"
+build_requires:
+  - CMake
+source: https://github.com/AliceO2Group/Common
+incremental_recipe: |
+  make ${JOBS:+-j$JOBS} install
+  mkdir -p $INSTALLROOT/etc/modulefiles && rsync -a --delete etc/modulefiles/ $INSTALLROOT/etc/modulefiles
+---
+#!/bin/sh
```
Comment (dberzano) [inline common-o2.sh]: Even if it is ignored, I'd say `#!/bin/bash -ex` which is what is actually being run.

File: infologger.sh [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  - Common-O2
+  - MySQL
+build_requires:
+  - CMake
+source: https://github.com/AliceO2Group/InfoLogger
+incremental_recipe: |
+  make ${JOBS:+-j$JOBS} install
+  mkdir -p $INSTALLROOT/etc/modulefiles && rsync -a --delete etc/modulefiles/ $INSTALLROOT/etc/modulefiles
+---
+#!/bin/sh
```
Comment (dberzano) [inline infologger.sh]: Same shebang note as above.

File: common-o2.sh (hunk 7 above)
Comment (Barthelemy) [inline common-o2.sh]: ok, will change. To be noted that o2 recipe also uses #!/bin/sh

File: common-o2.sh (hunk 1 above)
Comment (Barthelemy) [inline common-o2.sh]: ok, I will keep only BOOST_ROOT and use BOOST_VERSION

**Comment by Barthelemy** [comment]
@dberzano I think that I have adressed all your comments.

**Comment by Barthelemy** [comment]
@dberzano bump

**Pull request merged**